### PR TITLE
Record every payment attempt

### LINKED ISSUE #1422: Record payment method and failed payments
All payment transactions must have reference to used payment method (like it is done in auction) - so if used banklink or card payment it is clearly stated as well as if an invoice was marked as paid as a result of bank integration message or marked manually by administrator (in registry admin interface).

Furthermore all failed transactions should be recorded as well - ie failed card payments or bank link payments (for example due to not pressing back to merchant button - timeout).

### PR DESCRIPTION
**Pull request by karlerikounapuu**
Closes #1422

Records every attempt of paying invoice, including:
- banklinks (LHV, Swedbank, SEB),
- card payments (EveryPay)
- manual payments by admin, both autobind_invoices and targeted binding
- LHVConnect rake tast

... and keeps log of response and errors as in a way it's currently implemented in auction_center.

Rewritten logic when paying via gateways. Prior to this PR, BankTransaction was created after every attempt of paying invoice, regardless of it's outcome. It has been replaced with PaymentOrder and BankTransactions are created on-demand, eg when payment actually was, indeed, successful.

### CODE REVIEW

File: app/models/payment_order.rb [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+      class_name = ('PaymentOrders::' + method.camelize).constantize
+      raise(NoMethodError, class_name) unless class_name < PaymentOrder
+
+      supported << class_name
+    end
+
+    supported
+  end
+
+  def self.create_with_type(type:, invoice:)
```
Comment (yulgolem) [inline app/models/payment_order.rb]: In Rails way methods like #new usually generates new objects, while #create - generates and saves.
This method is called #create_*, but behaves like #new.

Please, consider renaming or changing the behaviour, as you refactors this - there is no need to drag old fallacies in a bright new future =).

File: app/models/payment_order.rb [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+  def complete_transaction
+    return NoMethodError unless payment_received?
+
+    paid!
+    transaction = composed_transaction
+    transaction.save! && transaction.bind_invoice(invoice.number)
+    return unless transaction.errors.any?
+
+    worded_errors = 'Failed to bind. '
```
Comment (yulgolem) [inline app/models/payment_order.rb]: Do we need to I18n that?

Comment (karlerikounapuu) [reply app/models/payment_order.rb]: I think it's not relevant, only shown to admin users anyway.

**Comment by vohmar** [comment]
migration error on deployment:

`fatal: [staging.internet.ee]: FAILED! => {"changed": true, "cmd": "RAILS_ENV='staging' rake db:migrate", "delta": "0:00:07.132846", "end": "2020-03-03 18:39:08.721087", "msg": "non-zero return code", "rc": 1, "start": "2020-03-03 18:39:01.588241", "stderr": "rake aborted!\nStandardError: An error has occurred, this and all later migrations canceled:\n\nPG::DuplicateTable: ERROR:  relation \"payment_orders\" already exists\n: CREATE TABLE \"payment_orders\" (\"id\" serial NOT NULL PRIMARY KEY, \"type\" character varying NOT NULL, \"status\" character varying DEFAULT 'issued' NOT NULL, \"invoice_id\" integer, \"response\" jsonb, \"notes\" character varying, \"creator_str\" character varying, \"updator_str\" character varying, \"created_at\" timestamp NOT NULL, \"updated_at\" timestamp NOT NULL, CONSTRAINT \"fk_rails_f9dc5857c3\"\nFOREIGN KEY (\"invoice_id\")\n  REFERENCES \"invoices\" (\"id\")\n)\n/home/registrant/deploy/releases/20200303183821/db/migrate/20200303120840_create_payment_orders.rb:3:in `change'\n\nCaused by:\nActiveRecord::StatementInvalid: PG::DuplicateTable: ERROR:  relation \"payment_orders\" already exists\n: CREATE TABLE \"payment_orders\" (\"id\" serial NOT NULL PRIMARY KEY, \"type\" character varying NOT NULL, \"status\" character varying DEFAULT 'issued' NOT NULL, \"invoice_id\" integer, \"response\" jsonb, \"notes\" character varying, \"creator_str\" character varying, \"updator_str\" character varying, \"created_at\" timestamp NOT NULL, \"updated_at\" timestamp NOT NULL, CONSTRAINT \"fk_rails_f9dc5857c3\"\nFOREIGN KEY (\"invoice_id\")\n  REFERENCES \"invoices\" (\"id\")\n)\n/home/registrant/deploy/releases/20200303183821/db/migrate/20200303120840_create_payment_orders.rb:3:in `change'\n\nCaused by:\nPG::DuplicateTable: ERROR:  relation \"payment_orders\" already exists\n/home/registrant/deploy/releases/20200303183821/db/migrate/20200303120840_create_payment_orders.rb:3:in `change'\nTasks: TOP => db:migrate\n(See full trace by running task with --trace)", "stderr_lines": ["rake aborted!", "StandardError: An error has occurred, this and all later migrations canceled:", "", "PG::DuplicateTable: ERROR:  relation \"payment_orders\" already exists", ": CREATE TABLE \"payment_orders\" (\"id\" serial NOT NULL PRIMARY KEY, \"type\" character varying NOT NULL, \"status\" character varying DEFAULT 'issued' NOT NULL, \"invoice_id\" integer, \"response\" jsonb, \"notes\" character varying, \"creator_str\" character varying, \"updator_str\" character varying, \"created_at\" timestamp NOT NULL, \"updated_at\" timestamp NOT NULL, CONSTRAINT \"fk_rails_f9dc5857c3\"", "FOREIGN KEY (\"invoice_id\")", "  REFERENCES \"invoices\" (\"id\")", ")", "/home/registrant/deploy/releases/20200303183821/db/migrate/20200303120840_create_payment_orders.rb:3:in `change'", "", "Caused by:", "ActiveRecord::StatementInvalid: PG::DuplicateTable: ERROR:  relation \"payment_orders\" already exists", ": CREATE TABLE \"payment_orders\" (\"id\" serial NOT NULL PRIMARY KEY, \"type\" character varying NOT NULL, \"status\" character varying DEFAULT 'issued' NOT NULL, \"invoice_id\" integer, \"response\" jsonb, \"notes\" character varying, \"creator_str\" character varying, \"updator_str\" character varying, \"created_at\" timestamp NOT NULL, \"updated_at\" timestamp NOT NULL, CONSTRAINT \"fk_rails_f9dc5857c3\"", "FOREIGN KEY (\"invoice_id\")", "  REFERENCES \"invoices\" (\"id\")", ")", "/home/registrant/deploy/releases/20200303183821/db/migrate/20200303120840_create_payment_orders.rb:3:in `change'", "", "Caused by:", "PG::DuplicateTable: ERROR:  relation \"payment_orders\" already exists", "/home/registrant/deploy/releases/20200303183821/db/migrate/20200303120840_create_payment_orders.rb:3:in `change'", "Tasks: TOP => db:migrate", "(See full trace by running task with --trace)"], "stdout": "== 20200303120840 CreatePaymentOrders: migrating ==============================\n-- adapter_name()\n   -> 0.0000s\n-- adapter_name()\n   -> 0.0000s\n-- create_table(:payment_orders, {:id=>:integer})", "stdout_lines": ["== 20200303120840 CreatePaymentOrders: migrating ==============================", "-- adapter_name()", "   -> 0.0000s", "-- adapter_name()", "   -> 0.0000s", "-- create_table(:payment_orders, {:id=>:integer})"]}`

**Pull request merged**